### ISSUE TITLE
For a `run` task, a value of `applicationDefaultJvmArgs` is lost, if jvm argument with `-javaagent` is added via init script

### ISSUE THREAD

**Issue by nskvortsov** [gradle/gradle#15239]
If an init script modifies jvm args of a `run` task by adding an instrumentation agent, default args configured in application plugin are lost (not passed to resulting jvm)

### Expected Behavior
jvm args, configured in `application.applicationDefaultJvmArgs` are available when executing `run` task

### Current Behavior
jvm args are lost.

### Context
This issue is reproducible using JDK15 (I was using Amazon Corretto-15)
Gradle version 6.7
Originally reported as an issue in Intellij IDEA [IDEA-255414](https://youtrack.jetbrains.com/issue/IDEA-255414), caused by Kotlin coroutines java agent, added by Kotlin IDEA plugin.

To me, it looks like all of sudden, a convention mapping of `JavaExec` instance gets broken and getJvmArgs() no longer takes `applicationDefaultJvmArgs` into account

### Steps to Reproduce 
PFA sample project: [defaultJvmArgs.zip](https://github.com/gradle/gradle/files/5568787/defaultJvmArgs.zip)
* Make sure, java is set to JDK15 (it does not reproduce on JDK11!)
* Unpack project
* run `gradlew build` - this will build a sample app and a sample no-op instrumenter
* run `gradlew run` - observe  (null value for `bar` is expected)
>foo=value
>bar=null

* run `gradlew run --init-script init.gradle`  (null value for `foo` is *not* expected)
>foo=null
>bar=null

setting `jvmArgs` to something other, than `-javaagent` does not break default args! 
* uncomment sample line in `init.gradle:4`, 
* comment out `-javaagent`  in `init.gradle:3`
* run `gradlew run --init-script init.gradle`  - observe (defaults are preserved!)
>foo=value
>bar=value

### Your Environment
Windows 10.0. amazon corretto-15, Gradle 6.7

**Comment by OliverO2** [comment]
In [KT-43453](https://youtrack.jetbrains.com/issue/KT-43453) there is now a sample project attached which reproduces with Gradle 6.8 on JDK 11 (corretto-11.0.11).

**Comment by OliverO2** [comment]
The issue is still current with Gradle 7.4.2. [See here for an updated reproducer](https://github.com/OliverO2/gradle-kotlin-jvmArgs).

**Comment by jasonsparc** [comment]
I'm just duplicating what I commented at #13463 (in case the following would be helpful).

---

@mlopatkin PR #23924 seems relevant to issue #15239

From what I can understand, the PR sets up `applicationDefaultJvmArgs` via the new `jvmArguments` as its `convention()` value. What this means is that, should `jvmArguments` be set to any other value, such as simply, using `jvmArguments.add()` for example, then what you end up getting is that `applicationDefaultJvmArgs` is now effectively lost **unintuitively**. Moreover, there's nothing in the docs that say that, `applicationDefaultJvmArgs` is only applied if you don't apply your own `jvmArgs`.

Also note that, `jvmArgs()` delegates to `jvmArguments.add()` now (as of Gradle 8.4 at least). See, https://github.com/gradle/gradle/blob/v8.4.0/platforms/jvm/language-java/src/main/java/org/gradle/api/tasks/JavaExec.java#L230

I don't feel like opening a new issue since I merely glanced at that PR and noticed this issue, so I could be wrong.

---

Also, regarding what I said about the docs about `applicationDefaultJvmArgs`, check out the following:

- https://docs.gradle.org/current/dsl/org.gradle.api.plugins.JavaApplication.html#org.gradle.api.plugins.JavaApplication:applicationDefaultJvmArgs
- https://docs.gradle.org/8.4/userguide/application_plugin.html#sec:application_extension:~:text=can%20configure%20the-,applicationDefaultJvmArgs,-property.%20These%20JVM

    Quote from the above link:

    > If your application requires a specific set of JVM settings or system properties, you can configure the `applicationDefaultJvmArgs` property. These JVM arguments are applied to the run task and also considered in the generated start scripts of your distribution.

**Comment by ljacomet** [comment]
Sorry for the late reply!

This issue needs a decision from the team responsible for that area. They have been informed. Response time may vary.

**Comment by big-guy** [comment]
This is blocked on how we plan to solve #18352
